test/framework/src/main/java/org/elasticsearch/test/rest/yaml/ESClientYamlSuiteTestCase.java

```diff
@@ -306,14 +306,31 @@ public void test() throws IOException {
 
         try {
             for (ExecutableSection executableSection : testCandidate.getTestSection().getExecutableSections()) {
-                executableSection.execute(restTestExecutionContext);
+                executeSection(executableSection);
             }
         } finally {
             logger.debug("start teardown test [{}]", testCandidate.getTestPath());
             for (DoSection doSection : testCandidate.getTeardownSection().getDoSections()) {
-                doSection.execute(restTestExecutionContext);
+                executeSection(doSection);
             }
             logger.debug("end teardown test [{}]", testCandidate.getTestPath());
         }
     }
+
+    /**
+     * Execute an {@link ExecutableSection}, careful to log its place of origin on failure.
+     */
+    private void executeSection(ExecutableSection executableSection) {
+        try {
+            executableSection.execute(restTestExecutionContext);
+        } catch (Exception e) {
+            throw new RuntimeException(errorMessage(executableSection, e), e);
+        } catch (AssertionError e) {
+            throw new AssertionError(errorMessage(executableSection, e), e);
```

Comment:
Sometimes it is really useful to figure out how the thing failed if it failed in a weird way. I'd say it isn't usually useful but it is nice to have all the time on the off chance you need it.